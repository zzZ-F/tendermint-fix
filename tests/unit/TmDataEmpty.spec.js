import { mount } from "@vue/test-utils"
import htmlBeautify from "html-beautify"
import { TmDataEmpty } from "@/index"

describe("TmDataEmpty", () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(TmDataEmpty)
  })

  it("has the expected html structure", () => {
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })

  it("has an icon", () => {
    expect(
      wrapper
        .find(".tm-data-msg__icon i.material-icons")
        .text()
        .trim()
    ).toBe("info_outline")
  })

  it("has a title", () => {
    expect(
      wrapper
        .find(".tm-data-msg__title h4")
        .text()
        .trim()
    ).toBe("N/A")
  })

  it("has a subtitle", () => {
    expect(
      wrapper
        .find(".tm-data-msg__subtitle h5")
        .text()
        .trim()
    ).toBe("No data available yet.")
  })
})
