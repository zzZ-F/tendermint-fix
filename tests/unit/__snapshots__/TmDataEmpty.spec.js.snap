// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmDataEmpty has the expected html structure 1`] = `
<div class="tm-data-msg">
  <div class="tm-data-msg__icon">
    <i class="material-icons">info_outline</i>
  </div>
  <div class="tm-data-msg__text">
    <div class="tm-data-msg__title">
      <h4>
        N/A
      </h4>
    </div>
    <div class="tm-data-msg__subtitle">
      <h5>
        No data available yet.
      </h5>
    </div>
  </div>
</div>
`;
